ansible/roles/csm.sbps.lio_config/files/provision_iscsi_server.sh

```diff
@@ -41,12 +41,17 @@ function add_server_target()
 {
         TARGET_SERVER_IQN="${IQN_PREFIX}$1"
         NMN_IP="$2"
-        HSN_IP="$3"
-        CMN_IP="$4"
+        CMN_IP="$3"
         targetcli "/iscsi create $TARGET_SERVER_IQN" &> /dev/null
         targetcli "/iscsi/${TARGET_SERVER_IQN}/tpg1/portals delete ip_address=0.0.0.0 ip_port=3260" &> /dev/null
         targetcli "/iscsi/${TARGET_SERVER_IQN}/tpg1/portals create ${NMN_IP}" &> /dev/null
-        targetcli "/iscsi/${TARGET_SERVER_IQN}/tpg1/portals create ${HSN_IP}" &> /dev/null
+
+        if [[ ! -z $HSN_IP ]]
```

Comment:
The arguments are all a bit convoluted.  This function is called with the global variables `$NMN_IP $CMN_IP $HSN_IP` as args 2-4, and then within this function we re-assign the global variables back to the same values.

Either have them be global variables and just reference them in this function without passing them in as arguments, or instead have them passed in and assign them to local variables used only within this function.  As it is now, it is confusing because the global variables are set before calling this function, and then set again to the same value in this function.